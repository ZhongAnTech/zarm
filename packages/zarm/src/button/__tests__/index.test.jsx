import React from 'react';
import { render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '../index';

describe('Button', () => {
  it('renders correctly', () => {
    const wrapper = render(<Button>foo</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('theme', () => {
    const wrapper = render(<Button theme="primary">foo</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('block', () => {
    const wrapper = render(<Button block>foo</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('ghost', () => {
    const wrapper = render(<Button ghost>foo</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('size', () => {
    const wrapper = render(<Button size="lg">foo</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('shape is radius', () => {
    const wrapper = render(<Button shape="radius">foo</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('icon', () => {
    const wrapper = render(
      <Button icon={<img alt="" src="https://zarm.design/images/logo.ce68565d.svg" />}>foo</Button>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('loading', () => {
    const wrapper = render(<Button loading>foo</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('disabled', () => {
    const wrapper = render(<Button disabled>foo</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('htmlType', () => {
    const wrapper = render(<Button htmlType="submit">foo</Button>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('onClick', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<Button onClick={onClick}>foo</Button>);
    wrapper.simulate('click');
    expect(onClick).toBeCalled();
  });

  it('onClick when disabled', () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <Button disabled onClick={onClick}>
        foo
      </Button>,
    );
    wrapper.simulate('click');
    expect(onClick).not.toBeCalled();
  });

  it('onTouchStart in NativeButton', () => {
    const wrapper = shallow(<Button>foo</Button>);
    wrapper.simulate('touchstart');
  });

  it('onTouchStart in AnchorButton', () => {
    const wrapper = shallow(<Button href="https://zarm.design">foo</Button>);
    wrapper.simulate('touchstart');
  });

  it('href and target', () => {
    const wrapper = render(
      <Button href="https://zarm.design" target="_blank">
        foo
      </Button>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
