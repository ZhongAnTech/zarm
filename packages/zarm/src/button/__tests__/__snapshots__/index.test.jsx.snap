// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button block 1`] = `
<button
  aria-disabled="false"
  class="za-button za-button--default za-button--md za-button--radius za-button--block"
  type="button"
>
  <span>
    foo
  </span>
</button>
`;

exports[`Button disabled 1`] = `
<button
  aria-disabled="true"
  class="za-button za-button--default za-button--md za-button--radius za-button--disabled"
  type="button"
>
  <span>
    foo
  </span>
</button>
`;

exports[`Button ghost 1`] = `
<button
  aria-disabled="false"
  class="za-button za-button--default za-button--md za-button--radius za-button--ghost"
  type="button"
>
  <span>
    foo
  </span>
</button>
`;

exports[`Button href and target 1`] = `
<a
  aria-disabled="false"
  class="za-button za-button--default za-button--md za-button--radius za-button--link"
  href="https://zarm.design"
  target="_blank"
>
  <span>
    foo
  </span>
</a>
`;

exports[`Button htmlType 1`] = `
<button
  aria-disabled="false"
  class="za-button za-button--default za-button--md za-button--radius"
  type="submit"
>
  <span>
    foo
  </span>
</button>
`;

exports[`Button icon 1`] = `
<button
  aria-disabled="false"
  class="za-button za-button--default za-button--md za-button--radius"
  type="button"
>
  <div
    class="za-button__content"
  >
    <img
      alt=""
      src="https://zarm.design/images/logo.ce68565d.svg"
    />
    <span>
      foo
    </span>
  </div>
</button>
`;

exports[`Button loading 1`] = `
<button
  aria-disabled="false"
  class="za-button za-button--default za-button--md za-button--radius za-button--loading"
  type="button"
>
  <div
    class="za-button__content"
  >
    <div
      class="za-activity-indicator za-activity-indicator--circular"
    >
      <svg
        viewBox="31 31 62 62"
      >
        <circle
          cx="62"
          cy="62"
          fill="none"
          r="28.5"
          style="stroke-width:5"
        />
      </svg>
    </div>
    <span>
      foo
    </span>
  </div>
</button>
`;

exports[`Button renders correctly 1`] = `
<button
  aria-disabled="false"
  class="za-button za-button--default za-button--md za-button--radius"
  type="button"
>
  <span>
    foo
  </span>
</button>
`;

exports[`Button shape is radius 1`] = `
<button
  aria-disabled="false"
  class="za-button za-button--default za-button--md za-button--radius"
  type="button"
>
  <span>
    foo
  </span>
</button>
`;

exports[`Button size 1`] = `
<button
  aria-disabled="false"
  class="za-button za-button--default za-button--lg za-button--radius"
  type="button"
>
  <span>
    foo
  </span>
</button>
`;

exports[`Button theme 1`] = `
<button
  aria-disabled="false"
  class="za-button za-button--primary za-button--md za-button--radius"
  type="button"
>
  <span>
    foo
  </span>
</button>
`;
